code/ui/Hud.cs

```diff
@@ -23,54 +19,100 @@ public Hud()
             }
 
-            Instance = this;
+            Current = this;
 
-            RootPanel.AddChild<ChatBox>();
-            RootPanel.AddChild<VoiceList>();
-            RootPanel.AddChild<GameTimer>();
-            RootPanel.AddChild<InfoFeed>();
-            RootPanel.AddChild<PostRoundMenu>();
-            RootPanel.AddChild<QuickShop>();
+            GeneralHudPanel = new GeneralHud(RootPanel);
+            AliveHudPanel = new AliveHud(RootPanel);
+        }
 
-            Scoreboard = RootPanel.AddChild<Scoreboard>();
+        [Event.Hotload]
+        public static void OnHotReloaded()
+        {
+            if (Host.IsClient)
+            {
+                Local.Hud?.Delete();
+                Hud hud = new Hud();
+                if (Local.Client.Pawn is TTTPlayer player && player.LifeState == LifeState.Alive)
+                {
+                    hud.AliveHudPanel.CreateHud();
+                }
+            }
         }
 
-        [Event("tttreborn.player.died")]
-        private void OnPlayerDied(TTTPlayer player)
+        [Event("tttreborn.player.spawned")]
+        private void OnPlayerSpawned(TTTPlayer player)
         {
             if (player != Local.Client.Pawn)
             {
                 return;
             }
 
-            _playerInfo?.Delete();
-            _playerInfo = null;
-
-            _weaponSelection?.Delete();
-            _weaponSelection = null;
-
-            _inspectMenu?.Delete();
-            _inspectMenu = null;
-
-            _nameplate?.Delete();
-            _nameplate = null;
+            Current?.AliveHudPanel.CreateHud();
         }
 
-        [Event("tttreborn.player.spawned")]
-        private void OnPlayerSpawned(TTTPlayer player)
+        [Event("tttreborn.player.died")]
+        private void OnPlayerDied(TTTPlayer player)
         {
             if (player != Local.Client.Pawn)
             {
                 return;
             }
 
-            _playerInfo ??= RootPanel.AddChild<PlayerInfo>();
+            Current?.AliveHudPanel.DeleteHud();
+        }
+
+        public class GeneralHud : Panel
+        {
+            public Scoreboard Scoreboard;
 
-            _weaponSelection ??= RootPanel.AddChild<WeaponSelection>();
+            public GeneralHud(Panel parent)
+            {
+                Parent = parent;
+
+                Parent.AddChild<ChatBox>();
+                Parent.AddChild<VoiceList>();
+                Parent.AddChild<GameTimer>();
+                Parent.AddChild<InfoFeed>();
+                Parent.AddChild<PostRoundMenu>();
+                Scoreboard = Parent.AddChild<Scoreboard>();
+            }
+        }
 
-            _inspectMenu ??= RootPanel.AddChild<InspectMenu>();
+        public class AliveHud : Panel
+        {
+            private PlayerInfo _playerInfo;
+            private WeaponSelection _weaponSelection;
+            private InspectMenu _inspectMenu;
+            private Nameplate _nameplate;
+            private QuickShop _quickShop;
```

Comment:
if we don't need to access any of them, we could put them in a list to avoid redundant code.